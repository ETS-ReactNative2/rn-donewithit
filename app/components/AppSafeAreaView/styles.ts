import { StyleSheet } from 'react-native'
import Constants from 'expo-constants'

/**
 * Constants.statusBarHeight
 * sdk_gphone64_x86_64: 24
 * M2102K1C: 39
 * IOS device: 44
 */

const styles = StyleSheet.create({
  container: {
    paddingTop: Constants.statusBarHeight,
    flex: 1
  }
})

export default styles
