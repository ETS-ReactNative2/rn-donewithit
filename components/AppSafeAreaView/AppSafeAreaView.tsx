import React from 'react'
import { SafeAreaView, StyleSheet } from 'react-native'
import Constants from 'expo-constants'

/**
 * Constants.statusBarHeight
 * sdk_gphone64_x86_64: 24
 * M2102K1C: 39
 * IOS device: 44
 */

const styles = StyleSheet.create({
  container: {
    paddingTop: Constants.statusBarHeight
  }
})

function AppSafeAreaView(props: any) {
  return (
    <SafeAreaView {...props} style={styles.container}>
      {props.children}
    </SafeAreaView>
  )
}

export default AppSafeAreaView
