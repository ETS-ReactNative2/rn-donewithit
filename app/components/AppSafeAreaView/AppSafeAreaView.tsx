import React from 'react'
import { SafeAreaView } from 'react-native'
import styles from './styles'

function AppSafeAreaView({ style, children, ...rest }: any) {
  return (
    <SafeAreaView style={[style, styles.container]} {...rest}>
      {children}
    </SafeAreaView>
  )
}

export default AppSafeAreaView
