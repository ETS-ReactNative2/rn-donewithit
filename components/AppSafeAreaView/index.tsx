import AppSafeAreaView from './AppSafeAreaView'

export default AppSafeAreaView
