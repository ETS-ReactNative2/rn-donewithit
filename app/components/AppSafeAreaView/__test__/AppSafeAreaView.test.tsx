import React from 'react'
import { StyleSheet } from 'react-native'
import renderer from 'react-test-renderer'
import AppSafeAreaView from '../index'

const customStyles = StyleSheet.create({
  container: {
    backgroundColor: 'red'
  }
})

const props = {
  style: customStyles.container
}

describe('AppSafeAreaView', () => {
  it('renders correctly', () => {
    const tree = renderer.create(<AppSafeAreaView {...props} />).toJSON()
    expect(tree).toMatchSnapshot()
  })
})
