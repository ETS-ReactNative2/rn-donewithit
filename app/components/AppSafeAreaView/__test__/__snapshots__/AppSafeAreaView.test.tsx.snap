// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`AppSafeAreaView renders correctly 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  style={
    Array [
      Object {
        "backgroundColor": "red",
      },
      Object {
        "flex": 1,
        "paddingTop": 20,
      },
    ]
  }
/>
`;
